team/challenges/systems/scheduler.tex

```diff
@@ -0,0 +1,274 @@
+
+    We appreciate your time and looking forward to hack on this little project together.
+
+\section{Objective}
+
+    Implement a part of a distributed job scheduler: a linux job worker server that executes arbitrary linux processes based on the API requests from clients.
+
+\section{Requirements} 
+
+  There are 6 engineering levels at Teleport. Check the engineering levels document for more details.
+  To qualify for a level, implement requirements for all levels up to the one you are interested in.
+
+  Start with a very brief google doc that covers the edge cases and design approach and post it to the slack channel. After the doc is approved, implement interfaces and an example program using the library.
+
+  Add a couple of high quality tests that cover happy and unhappy scenarios.
+
+  Split the submission in 2-3 pull requests for us to review. We will review every pull request and provide our feedback.
+
+  We are going to compile the program, test it and get back to you.
+  
+  \subsection*{Level 1}
+
+  \subsubsection*{Library}
+  \begin{itemize} 
+  \item[] Worker library with methods to start/stop/query status and get an output of a running job.
+  \end{itemize}
+
+  \subsubsection*{API}
+  \begin{itemize} 
+  \item[] Add HTTPS API to start/stop/get status of a running process. Use basic authentication.
+  \end{itemize}
+
+  \subsubsection*{Client}
+  \begin{itemize} 
+  \item[] Client command should be able to connect to worker service and schedule several jobs.
+  \item[] Client should be able to query result of the job execution and fetch the logs
+  \end{itemize}    
+    
+ \subsection*{Level 2}
+
+ \subsubsection*{API}
+ \begin{itemize} 
+ \item[] Use mTLS instead of basic authentication and verify client certificate. Set up strong set of cipher suites for TLS and good crypto setup for certificates.
+ \end{itemize}
+
+ \subsection*{Level 3}
+    
+  \subsubsection*{API}
+  \begin{itemize} 
+  \item[] Use \href{https://grpc.io}{GRPC} for API and add a streaming output of a running job process.
+  \item[] Add streaming log output of a running job process.
+  \end{itemize}
+
+  \subsubsection*{Client}
+  \begin{itemize} 
+  \item[] Client should be able to stream the logs.
+  \end{itemize}
+  
+  \subsection*{Level 4}
+    
+  \subsubsection*{Resource control}
+  \begin{itemize}
+  \item[] Add limits for amount of CPU, Memory and Disk IO using system calls.
+  \end{itemize}
+
+  \subsection*{Level 5}
+    
+  \subsubsection*{Isolation}
+  \begin{itemize}
+  \item[] Each job process should be given a separate PID, Mount and Networking namespace using system calls.
+  \end{itemize}
+
+\section{Guidance}
+
+\subsection{Interview process}
+
+The interview team joins the slack channel. The team consists of the engineers who will be working with you.
+Ask them bout the engineering culture, work and life balance, or anything else that you would like to learn about Teleport.

```

Comment:
**a**bout

Reply:
I think it's spelled about just as you typed?

Reply:
```suggestion
Ask them about the engineering culture, work and life balance, or anything else that you would like to learn about Teleport.
```